includes/Core/Modules/Modules.php

```diff
@@ -244,6 +245,44 @@ function ( Module $module ) {
 			}
 		);
 
+		add_action(
+			'googlesitekit_authorize_user',
+			function( $token_response ) {
+				if ( empty( $token_response['analytics_configuration'] ) ) {
+					return;
+				}
+
+				// Do nothing if the Analytics module is already activated.
+				if ( $this->is_module_active( Analytics::MODULE_SLUG ) ) {
+					return;
+				}
+
+				$this->activate_module( Analytics::MODULE_SLUG );
+
+				$extra_scopes = $this->user_options->get( OAuth_Client::OPTION_ADDITIONAL_AUTH_SCOPES );
+				if ( is_array( $extra_scopes ) && in_array( Analytics::READONLY_SCOPE, $extra_scopes, true ) ) {
+					unset( $extra_scopes[ array_search( Analytics::READONLY_SCOPE, $extra_scopes, true ) ] );
```

Comment:
Nit-pick: We could only call `array_search` instead of both `array_search` and `in_array` here, and the condition could then just check if the found index is `> -1`. But nothing blocking.

Reply:
Ok, updated.